safir/src/safir/metrics/_event_manager.py

```diff
@@ -155,215 +374,176 @@ class MyEvent(EventPayload):
+            calling this method.
         """
         if not self._initialized:
-            raise EventManagerUnintializedError(
-                "EventManager instance must be initialized before creating"
-                " event publishers"
-            )
-
-        time_ns = time.time_ns()
-        metadata = EventMetadata(
-            id=uuid4(),
-            app_name=self._app_name,
-            timestamp=self._ns_to_datetime(time_ns),
-            timestamp_ns=time_ns,
-        )
-        event = event_class(**metadata.model_dump(), **payload.model_dump())
-
-        if self._disable:
-            self._logger.debug(
-                "Would have published event",
-                metrics_event=event.model_dump(),
-            )
-            return event
-
+            msg = "Initialize EventManager before creating event publishers"
+            raise EventManagerUnintializedError(msg)
         encoded = await self._schema_manager.serialize(event)
         await publisher.publish(encoded)
-
         self._logger.debug(
             "Published metrics event",
             metrics_event=event.model_dump(),
             topic=publisher.topic,
             schema_info=schema_info,
         )
 
-        return event
+    async def _is_topic_ok(self, topic: str) -> bool:
+        """Check whether the given Kafka topic is valid.
 
-    def _ns_to_datetime(self, ns: int) -> datetime:
-        """Convert a int number of nanoseconds to a datetime."""
-        return datetime.fromtimestamp(ns / 1e9, tz=UTC)
+        Parameters
+        ----------
+        topic
+            Kafka topic to check.
+
+        Returns
+        -------
+        bool
+            `True` if the Kafka topic is valid, `False` otherwise.
+        """
+        info = await self._admin_client.describe_topics([topic])
+        if len(info) == 0:
+            return False
+        topic_info = info[0]
+        error_code = topic_info.get("error_code", None)
+        return error_code == 0
+
+
+class NoopEventManager(EventManager):
+    """An event manager that creates publishers that quietly do nothing.
+
+    This is used as the implementation of `~safir.metrics.EventManager` when
+    event publication is disabled. The event type registrations and event
+    payloads are still verified to catch errors, but are then discarded.
+
+
+    Parameters
+    ----------
+    application
+        Name of the application that is generating events.
+    topic_prefix
+        Kafka topic prefix for the metrics events topic for this application.
+    logger
+        Logger to use for internal logging.
+    """
+
+    def __init__(
+        self,
+        application: str,
+        topic_prefix: str,
+        logger: BoundLogger | None = None,
+    ) -> None:
+        super().__init__(f"{topic_prefix}.{application}")
+        self._application = application
+        self._logger = logger or structlog.get_logger("safir.metrics")
```

Comment:
As long as you're putting some of the common `__init__` logic in the base class, why not put this there too?

Reply:
"Because it's more clear what's going on" is a perfectly valid answer, and maybe why my version of this was so complex

Reply:
Yeah, this is true -- it's because if I put it in the parent class, I have to make logger a public attribute, which feels weird because things outside of the event manager shouldn't really use it. Or I leave it as a private attribute and then subclasses have to know they can use the private attribute, so it breaks the encapsulation. I sometimes make weird choices because Python doesn't really have a convention for protected attributes that are intended for subclasses but not general use. I should get over it and just move the common code. :) I'll do that.